tests/pipelines/earliest.nf.test

```diff
@@ -288,4 +288,79 @@ nextflow_pipeline {
+            assert transformation.text.contains("sample3,2000-03-01")
+            assert transformation.text.contains("sample4,")
+            assert transformation.text.contains("sample5,")
+            assert transformation.text.contains("sample6,")
+            assert transformation.text.contains("sample7,")
+
+            // Check Results (Human-Readable)
+            def results = path("$launchDir/results/transformation/results.csv")
+            assert results.exists()
+
+            assert results.text.contains("sample,sample_name,metadata_1,metadata_2,metadata_3,metadata_4,metadata_5,metadata_6,metadata_7,metadata_8,earliest_date,earliest_date_valid,earliest_date_error")
+            assert results.text.contains("sample1,normal,2000-01-01,Not Applicable,Missing,Not Collected,Not Provided,Restricted Access,,2000-01-08,2000-01-01,True,")
+            assert results.text.contains("sample2,mixed_case,2000-02-01,not Applicable,MIssIng,not Collected,not Provided,restricted Access,,2000-02-08,2000-02-01,True,")
+            assert results.text.contains("sample3,lowercase,2000-03-01,not applicable,missing,not collected,not provided,restricted access,,2000-03-08,2000-03-01,True,")
+            assert results.text.contains("sample4,bad_data,2000-03-01,not applicable,BAD_DATA,not collected,not provided,restricted access,,2000-03-08,,False,At least one of the dates are incorrectly formatted.")
+            assert results.text.contains("sample5,bad_date_format,2000-03-01,not applicable,2000-20-01,not collected,not provided,restricted access,,2000-03-08,,False,At least one of the dates are incorrectly formatted.")
+            assert results.text.contains("sample6,all_special,not applicable,not applicable,missing,not collected,not provided,restricted access,not applicable,not applicable,,False,No dates were found.")
+            assert results.text.contains("sample7,all_special_and_blank,,not applicable,missing,not collected,not provided,restricted access,,,,False,No dates were found.")
+
+            // Check IRIDA Next JSON Output
+            def iridanext_json = path("$launchDir/results/iridanext.output.json").json
+            def iridanext_global = iridanext_json.files.global
+            def iridanext_metadata = iridanext_json.metadata.samples
+
+            assert iridanext_global.findAll { it.path == "transformation/results.csv" }.size() == 1
+
+            assert iridanext_metadata.size() == 7
+
+            assert iridanext_metadata.containsKey("sample1")
+            assert iridanext_metadata.sample1.earliest_date == "2000-01-01"
+
+            assert iridanext_metadata.containsKey("sample2")
+            assert iridanext_metadata.sample2.earliest_date == "2000-02-01"
+
+            assert iridanext_metadata.containsKey("sample3")
+            assert iridanext_metadata.sample3.earliest_date == "2000-03-01"
+
+            assert iridanext_metadata.containsKey("sample4")
+            assert iridanext_metadata.sample4.earliest_date == ""
```

Comment:
Having an empty string/value for `earliest_date` in the IRIDA Next JSON file means that whatever the previous value for `earliest_date` will be deleted from IRIDA Next. I will have to check in if this is the expected behaviour.

Reply:
Based on discussions with others, we don't want the pipeline to overwrite `earliest_date` for a sample that was passed invalid date. So, in that case, the logic would need to change to remove the empty strings/values for sample4 to sample7 in this particular test case from the IRIDA Next JSON file.

Reply:
Update in cb0cd52